### PR TITLE
Remove all pre 2.1 logic

### PR DESCRIPTION
**Pull request by ptnapoleon**
So I will clearly set up and run a CI job with my changes, but looking for review from @mambocab and @EnigmaCurry.

### CODE REVIEW

**Comment by EnigmaCurry** [comment]
This is awesome.

**Comment by ptnapoleon** [comment]
For those unaware, as 2.0 and earlier are EOL, and we aren't testing them anymore, there's no need to clutter the codebase with any pre-2.1 logic.

**Comment by knifewine** [comment]
Nice.

**Comment by mambocab** [comment]
I've only given this a quick skim so far, but very exciting stuff. Looks like there are a bunch of `since('2.1')`s hanging around too. Any reason to keep them around?

**Comment by ptnapoleon** [comment]
Nope, I'll get those too

File: commitlog_test.py [hunk 1]
```diff
@@ -478,11 +471,7 @@ def get_header_crc(header):
             # read the header and find the crc location
             with open(os.path.join(cl_dir, cl), 'r') as f:
                 f.seek(0)
-                version = struct.unpack('>i', f.read(4))[0]
                 crc_pos = 12
-                # if version < 5:
```
Comment (mambocab) [inline commitlog_test.py]: If this is removed, we may also be able to take out this:

https://github.com/riptano/cassandra-dtest/blob/two-oh-cleanup/commitlog_test.py#L388

Maybe check with Blake, I don't know what the logic is.

Comment (mambocab) [inline commitlog_test.py]: btw this isn't a blocker for merging this, just something to clean up if we get the chance/it's easy.

**Comment by mambocab** [comment]
Review by visual inspection looks good so far (as of 5e7ddb9).

**Comment by ptnapoleon** [comment]
The run with 5e7ddb9 succeeded on cassci. Waiting on the removal of the since code now.

**Comment by ptnapoleon** [comment]
Okay the results look great: http://cassci.datastax.com/job/ptnapoleon-2.0-cleanup-dtest/

Will squash/merge after a +1

**Comment by mambocab** [comment]
+1 post-squash as well.

**Comment by ptnapoleon** [comment]
\o/. Thanks

**Pull request merged**